src/extrinsic/mod.rs

```diff
@@ -17,10 +17,46 @@
 
 //! Offers some predefined extrinsics for common runtime modules.
 
+pub use balances::CreateBalancesExtrinsic;
+pub use contracts::CreateContractsExtrinsic;
+#[cfg(feature = "staking-xt")]
+pub use staking::CreateStakingExtrinsic;
+pub use utility::CreateUtilityExtrinsic;
+
 pub mod balances;
-pub mod common;
 pub mod contracts;
 pub mod offline_extrinsic;
 #[cfg(feature = "staking-xt")]
 pub mod staking;
 pub mod utility;
+
+use crate::Api;
+use ac_primitives::{ExtrinsicParams, FrameSystemConfig, SignExtrinsic, UncheckedExtrinsicV4};
+
+pub type AddressFor<Module> = <Module as AssignExtrinsicTypes>::Address;
+pub type SignatureFor<Module> = <Module as AssignExtrinsicTypes>::Signature;
+pub type SignedExtraFor<Module> = <Module as AssignExtrinsicTypes>::SignedExtra;
+
+/// Type to summarize and simplify the extrinsic return values. It defines all the types in
+/// UncheckedExtrinsicV4 except for the Call, which differs for every extrinsic.
+pub type ExtrinsicFor<Module, Call> =
+	UncheckedExtrinsicV4<AddressFor<Module>, Call, SignatureFor<Module>, SignedExtraFor<Module>>;
+
+/// Helper trait to assign the types used in every extrinsic.
+/// This helps to avoid repeated reassignments in the extrinsic creation traits.
+pub trait AssignExtrinsicTypes {
+	type Address;
+	type Signature;
+	type SignedExtra;
+}
+
+impl<Signer, Client, Params, Runtime> AssignExtrinsicTypes for Api<Signer, Client, Params, Runtime>
+where
+	Signer: SignExtrinsic<Runtime::AccountId>,
+	Runtime: FrameSystemConfig,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+{
+	type Address = Signer::ExtrinsicAddress;
+	type Signature = Signer::Signature;
+	type SignedExtra = Params::SignedExtra;
+}
```

Comment:
I think we don't need this trait if we have the Extrinsic associated types as I described above.

Reply:
I don't see the correlation : This trait is just so that not every trait needs to redefine these three types. Nothing to do with the Extrinsic I think?

Reply:
Nvm - there is a correlation.

Reply:
I thought `Extrinsic` is mostly where we use them as the trait is called `AssignExtrinsicTypes`, but you can quite conveniently have an extrinsic associated type to assign those values.

If this this re-assignment is needed beyond the scope of extrinsics, we should add these associated types to an overarching `ApiTrait` IMO.

Reply:
I removed it. If it's about the re-assignment, as you wrote, it should be done a level above.